installers-conda/build_installers.py

```diff
@@ -5,523 +5,331 @@
+
+
+def _output_file(install_type):
+    return DIST / f"Spyder-{OS}-{ARCH}.{install_type}"
+
+
+def _constructor(install_type, spy_ver, debug=False):
+    """Build installer from construct.yaml"""
+    DIST.mkdir(exist_ok=True)
 
     cmd_args = [
-        constructor, "-v",
+        "constructor",
         "--output-dir", str(DIST),
-        "--platform", TARGET_PLATFORM,
-        str(CONSTRUCTOR_FILE.parent)
+        "--platform", PLATFORM,
+        str(RESOURCES)
     ]
-    if args.debug:
+    if debug:
         cmd_args.append("--debug")
-    conda_exe = os.getenv("CONSTRUCTOR_CONDA_EXE")
-    if conda_exe:
-        cmd_args.extend(["--conda-exe", conda_exe])
 
     env = os.environ.copy()
-    env["CONDA_CHANNEL_PRIORITY"] = "flexible"
+    env.update(
+        {
+            "OS": OS,
+            "ARCH": ARCH,
+            "INSTALL_TYPE": install_type,
+            "INSTALL_VER": spy_ver,
+            "REPO_PATH": str(SPYREPO),
+        }
+    )
 
     logger.info("Command: " + " ".join(cmd_args))
 
     run(cmd_args, check=True, env=env)
+    logger.info(f"Created {_output_file(install_type)}")
 
 
-def licenses():
-    info_path = BUILD / "info.json"
-    try:
-        info = json.load(info_path)
-    except FileNotFoundError:
-        print(
-            "!! Use `constructor --debug` to write info.json and get licenses",
-            file=sys.stderr,
-        )
-        raise
-
-    zipname = BUILD / f"licenses.{OS}-{ARCH}.zip"
-    output_zip = zipfile.ZipFile(zipname, mode="w",
-                                 compression=zipfile.ZIP_DEFLATED)
-    output_zip.write(info_path)
-    for package_id, license_info in info["_licenses"].items():
-        package_name = package_id.split("::", 1)[1]
-        for license_type, license_files in license_info.items():
-            for i, license_file in enumerate(license_files, 1):
-                arcname = (f"{package_name}.{license_type.replace(' ', '_')}"
-                           f".{i}.txt")
-                output_zip.write(license_file, arcname=arcname)
-    output_zip.close()
-    return zipname.resolve()
-
-
-def main():
+def _cleanup_build(debug=False):
+    if debug or not BUILD.exists():
+        # Do not clean the build directory
+        return
+
+    exts = (".exe", ".sh", ".pkg", ".yml", ".rtf", ".lock", ".png")
+    for f in BUILD.glob("*"):
+        if f.suffix in exts:
+            f.unlink()
+
+
+def main(spy_ver, extra_specs, install_type, no_local, debug):
+    BUILD.mkdir(exist_ok=True)
+    DIST.mkdir(exist_ok=True)
+
+    _cleanup_build()
+
+    _generate_background_images(install_type)
+
     t0 = time()
     try:
-        DIST.mkdir(exist_ok=True)
-        _create_conda_lock(env_type='base')
-        assert BASE_LOCK_FILE.exists()
-        _create_conda_lock(env_type='runtime')
-        assert RT_LOCK_FILE.exists()
+        _create_conda_lock('base', no_local=no_local)
+        _create_conda_lock('runtime', extra_specs, no_local)
```

Comment:
I guess you're planning to add the Spyder-updater env to the installers in a follow-up PR, right?

Reply:
Correct.